include/ginkgo/core/log/batch_logger.hpp

```diff
@@ -33,39 +33,12 @@ namespace detail {
 template <typename ValueType>
 struct log_data final {
     using real_type = remove_complex<ValueType>;
+    using idx_type = int;
```

Comment:
I wouldn't mind
```suggestion
    using index_type = int;
```
I think that would be more consistent with our naming scheme.